text/0000-guaranteed-slice-repr.md

```diff
@@ -0,0 +1,167 @@
+- Feature Name: guaranteed_slice_repr
+- Start Date: 2025-02-18
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC guarantees the in-memory representation of slice and str references.
+Specifically, `&[T]` is guaranteed to have the same layout as:
```

Comment:
this should probably state that it's not specifying the function call ABI, so `extern "C" fn(&[u8]) -> &[u8]` still doesn't have an ABI guaranteed to match passing that `repr(C)` struct by value